### PR TITLE
DBCON-283:  avoid transitive packages exposure.

### PR DESCRIPTION
**Pull request by nicomz**

### CODE REVIEW

File: src/main/java/org/mule/extension/db/internal/DbConnector.java [hunk 1]
```diff
@@ -7,14 +7,9 @@
...
 
 import org.mule.db.commons.AbstractDbConnector;
-import org.mule.db.commons.api.exception.connection.ConnectionCreationException;
 import org.mule.db.commons.api.exception.connection.DbError;
-import org.mule.db.commons.api.param.BulkQueryDefinition;
-import org.mule.db.commons.api.param.QueryDefinition;
-import org.mule.db.commons.api.param.StoredProcedureCall;
 import org.mule.db.commons.internal.DbFunctions;
 import org.mule.db.commons.internal.exception.DbExceptionHandler;
-import org.mule.extension.db.api.logger.LoggerApiPackage;
```
Comment (juldiazmule) [inline src/main/java/org/mule/extension/db/internal/DbConnector.java:17]: Shouldn't we continue to export LoggerApiPackage for MuleMySqlLogger?

**Review by estebanwasinger — APPROVED** [review]

File: src/main/java/org/mule/extension/db/internal/domain/connection/datasource/DbDataSourceReferenceConnectionProvider.java [hunk 2]
```diff
@@ -49,7 +48,7 @@
 @Alias("data-source")
 @ExternalLib(name = "JDBC Driver", description = "A JDBC driver which supports connecting to the Database",
     nameRegexpMatcher = DRIVER_FILE_NAME_PATTERN, type = JAR, optional = true)
-public class DbDataSourceReferenceConnectionProvider implements ConnectionProvider<DbConnection>, Initialisable, Disposable {
+public class DbDataSourceReferenceConnectionProvider extends DbConnectionProvider {
```
Comment (marianogonzalez) [inline src/main/java/org/mule/extension/db/internal/domain/connection/datasource/DbDataSourceReferenceConnectionProvider.java:51]: I don't see a reason why the client cannot contain a `DatasourceReferenceConnectionProvider` implementation and the connector simply extends it adding this annotations. This class is not exporting anything.

File: src/main/java/org/mule/extension/db/internal/domain/connection/derby/DerbyConnectionProvider.java [hunk 3]
```diff
@@ -91,63 +88,37 @@
...
-          return java.util.Optional.of(CANNOT_REACH);
-        }
-
-        return empty();
-      }
-
-    };
-
-    dbConnectionProvider.initialise();
+    super.columnTypes = columnTypes;
```
Comment (marianogonzalez) [inline src/main/java/org/mule/extension/db/internal/domain/connection/derby/DerbyConnectionProvider.java:93]: this seems unnecesary. The SDK will pick up the `@Parameters` in the super class

File: src/main/java/org/mule/extension/db/internal/domain/connection/generic/DbGenericConnectionProvider.java [hunk 4]
```diff
@@ -47,7 +45,7 @@
 @Alias("generic")
 @ExternalLib(name = "JDBC Driver", description = "A JDBC driver which supports connecting to the Database",
     nameRegexpMatcher = DRIVER_FILE_NAME_PATTERN, type = JAR)
-public class DbGenericConnectionProvider implements ConnectionProvider<DbConnection>, Initialisable, Disposable {
+public class DbGenericConnectionProvider extends DbConnectionProvider {
```
Comment (marianogonzalez) [inline src/main/java/org/mule/extension/db/internal/domain/connection/generic/DbGenericConnectionProvider.java:48]: same as before

File: src/main/java/org/mule/extension/db/internal/domain/connection/mysql/MySqlConnectionProvider.java [hunk 5]
```diff
@@ -92,62 +88,37 @@
   @ParameterGroup(name = CONNECTION)
   private MySqlConnectionParameters mySqlParameters;
```
Comment (marianogonzalez) [inline src/main/java/org/mule/extension/db/internal/domain/connection/mysql/MySqlConnectionProvider.java:89]: same as before

**Review by marianogonzalez — CHANGES_REQUESTED** [review]

**Review by marianogonzalez — APPROVED** [review]

**Pull request merged**